src/EditorFeatures/CSharp/QuickInfo/SyntacticQuickInfoProvider.cs

```diff
@@ -76,8 +85,53 @@ internal class SyntacticQuickInfoProvider : AbstractQuickInfoProvider
                 return null;
             }
 
-            var span = new SnapshotSpan(textSnapshot, Span.FromBounds(parent.SpanStart, openBrace.Span.End));
+            var span = new SnapshotSpan(textSnapshot, Span.FromBounds(spanStart, spanEnd));
             return this.CreateElisionBufferDeferredContent(span);
         }
+
+        private static void MarkInterestedSpanNearbyScopeBlock(SyntaxNode block, SyntaxToken openBrace, ref int spanStart, ref int spanEnd)
+        {
+            SyntaxTrivia nearbyComment;
+
+            var searchListAbove = openBrace.LeadingTrivia.Reverse();
+            if (FindFurthestNearbyComment(ref searchListAbove, out nearbyComment))
+            {
+                spanStart = nearbyComment.SpanStart;
+                return;
+            }
+
+            var nextToken = block.FindToken(openBrace.FullSpan.End);
+            var searchListBelow = nextToken.LeadingTrivia;
+            if (FindFurthestNearbyComment(ref searchListBelow, out nearbyComment))
+            {
+                spanEnd = nearbyComment.Span.End;
+                return;
+            }
+        }
+
+        private static bool FindFurthestNearbyComment<T>(ref T triviaSearchList, out SyntaxTrivia nearbyTrivia)
```

Comment:
Interesting.  Does this work?  You're going to get specializations for each T.  And I suppose these will be constrained interface calls through IEnumerable, which the JIT can optimize.  So, if this works, very cute and nifty :-)